reactfx/src/main/java/org/reactfx/value/Val.java

```diff
@@ -745,6 +748,60 @@ protected T computeValue() {
                 .flatMap(Window::showingProperty)
                 .orElseConst(false);
     }
+    
+    /**
+     * Returns a {@linkplain Val} with the value returned by {@code supplier}.
+     * The {@linkplain Supplier} will only be called when {@code getValue()} is
+     * called for the first time. It will not be called again. Although
+     * technically the value is undefined until requested for the first time,
+     * this will not result in a change. No listener will ever be notified.
+     */
+    static <T> Val<T> lazy(Supplier<T> supplier) {
+        return new RigidVal<T>() {
+            Supplier<T> sup = supplier;
+            T value;
+            
+            @Override
+            public T getValue() {
+                if (sup != null) {
+                    value = sup.get();
+                    sup = null;
+                }
+                return value;
+            }
+        };
+    }
+    
+    /**
+     * Returns a {@linkplain Val} whose value is {@code placeholder}. Upon the
+     * first call of {@code getValue()}, {@code callable} will be executed in
+     * parallel and the result will become the new value of this
+     * {@linkplain Val}. In contrast to {@linkplain Val#lazy(Supplier)}, this
+     * will result in listeners being notified.
+     * <p>
+     * The {@linkplain Callable} should handle any errors itself. Any
+     * {@linkplain Exception} will be rethrown on the FX Thread.
+     */
+    static <T> Val<T> lazyAsync(T placeholder, Callable<T> callable) {
+        // Not sure about Callable, as it specifically allows throwing any
+        // checked exception
+        return null;
+    }
+    
+    static <T> Val<T> lazyAsync(
+            T placeholder,
+            Callable<T> callable,
+            Executor executor) {
+        return null;
+    }
+    
+    static <T> Val<T> lazyAsync(
+            T placeholder,
+            Supplier<CompletionStage<T>> completionStage) {
```

Comment:
Maybe don't make this one lazy and accept a `CompletionStage` whose computation already started. Perhaps call it `awaitAsync`?